DeviceHandlerToken.h

```diff
@@ -0,0 +1,30 @@
+#ifndef THETA_CLOUD_DEVICE_HANDLER_TOKEN_H
+#define THETA_CLOUD_DEVICE_HANDLER_TOKEN_H
+
+#include <memory>
+#include <functional>
+
+class ThetaCloud;
+
+class DeviceHandlerToken
+{
+    DeviceHandlerToken() = delete;
+public:
+    ~DeviceHandlerToken()
+    {
+        deleter();
+    }
+
+protected:
+    DeviceHandlerToken(const std::function<void()> &deleter) :
+        deleter(deleter)
+    {
+    }
+
+    const std::function<void()> deleter;
```

Comment:
Maybe instead of std::function you can use function pointer?
Like:
```
using FunctionPtr = void(*)();
const FunctionPtr deleter;
```
std::function adds bloat, function ptr is faster, I've created a benchmark to show it: http://quick-bench.com/eDyrxB6vGZWQ0IUrPMTSkMSxSsA

Reply:
I've used `std::function` as I am passing lambdas there. I think I should store these lambdas somewhere as the pointers would get invalid after the lambda would get out of scope, so it would complicate the code quite much - or am I missing something?